### PR TITLE
Fix test linkage for hdk

### PR DESCRIPTION
**Pull request by leshikus**
Subj

### CODE REVIEW

File: Tests/ArrowSQLRunner/CMakeLists.txt [hunk 1]
```diff
@@ -5,5 +5,5 @@ set(arrow_query_runner_files
 )
 
 add_library(ArrowQueryRunner ${arrow_query_runner_files})
-target_link_libraries(ArrowQueryRunner gtest ${Boost_LIBRARIES})
+target_link_libraries(ArrowQueryRunner gtest ${Boost_LIBRARIES} QueryEngine ArrowStorage Calcite)
```
Comment (kurapov-peter) [inline Tests/ArrowSQLRunner/CMakeLists.txt:8]: Library dependencies with this signature are transitive by default. Was that your intention? If so, linking tests against `ArrowQueryRunner` does not require explicit `QueryEngine` dependency (for example in `target_link_libraries(ResultSetTest gtest QueryEngine ArrowQueryRunner ArrowStorage)`).
In general though, I think it's better to state the scope explicitly for library targets.

Comment (leshikus) [reply Tests/ArrowSQLRunner/CMakeLists.txt:8]: 1. In HDK build `QueryEngine` and `Calcite` was not added by default, so I've added it. I've added two `Arrow*` libraries to maintain the same style. 

2. IMHO  `EXECUTE_TEST_LIBS` name is misleading. It does not execute anything.

3. Putting all the libraries in one variable produces mess. `fmt::fmt ${Arrow_LIBRARIES} ${CMAKE_DL_LIBS} ${CUDA_LIBRARIES} ${Boost_LIBRARIES} ${ZLIB_LIBRARIES}` were not actually needed

4. We discussed a half a year ago how to split tests in to groups. Now the file gives an idea how to do it - it is visible which tests do test QueryEngine, and which test ArrowStorage. It also indicates which tests should be fixed to get rid of `Logger`.

Comment (leshikus) [reply Tests/ArrowSQLRunner/CMakeLists.txt:8]: BTW, there was a problem I was fighting for some time to install `fmt` from conda, and in fact it is not needed.

Comment (kurapov-peter) [reply Tests/ArrowSQLRunner/CMakeLists.txt:8]: My point was only about your current changes. The `ArrowQueryRunner` as created now already exports gtest, boost, query engine, storage, and calcite. Adding those as explicit dependencies to tests is just redundant.

**Review by kurapov-peter — APPROVED** [review]

**Comment by leshikus** [comment]
@Garra1980 asked me to add @alexbaden to the list of approvers.

Here is the patch I use to reproduce the problems in HDK:
```
diff --git a/CMakeLists.txt b/CMakeLists.txt
index 0849c48..ddd5a47 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -218,3 +218,47 @@ add_custom_target(hdk_python_clean

 add_dependencies(clean-all hdk_python_clean)
 add_dependencies(clean-all calcite_java_clean)
+
+# address and thread sanitizer
+option(ENABLE_STANDALONE_CALCITE "Require standalone Calcite server" OFF)
+option(ENABLE_ASAN "Enable address sanitizer" OFF)
+option(ENABLE_TSAN "Enable thread sanitizer" OFF)
+option(ENABLE_UBSAN "Enable undefined behavior sanitizer" OFF)
+if(ENABLE_ASAN)
+  set(SAN_FLAGS "-fsanitize=address -O1 -fno-omit-frame-pointer")
+  add_definitions("-DWITH_DECODERS_BOUNDS_CHECKING")
+elseif(ENABLE_TSAN)
+  add_definitions("-DHAVE_TSAN")
+  # Copy the config directory to the build dir for TSAN suppressions
+  file(COPY config DESTINATION ${CMAKE_BINARY_DIR})
+
+  set(SAN_FLAGS "-fsanitize=thread -fPIC -O1 -fno-omit-frame-pointer")
+  # required for older GCC, see https://gcc.gnu.org/bugzilla/show_bug.cgi?id=64354
+  add_definitions("-D__SANITIZE_THREAD__")
+elseif(ENABLE_UBSAN)
+  set(SAN_FLAGS "-fsanitize=undefined -fPIC -O1 -fno-omit-frame-pointer")
+endif()
+if(ENABLE_ASAN OR ENABLE_TSAN OR ENABLE_UBSAN)
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${SAN_FLAGS}")
+  set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${SAN_FLAGS}")
+  set(ENABLE_STANDALONE_CALCITE ON)
+endif()
+
+option(ENABLE_DECODERS_BOUNDS_CHECKING "Enable bounds checking for column decoding" OFF)
+
+if(ENABLE_STANDALONE_CALCITE)
+  add_definitions("-DSTANDALONE_CALCITE")
+endif()
+
+option(ENABLE_TESTS "Build unit tests" ON)
+if (ENABLE_TESTS)
+  enable_testing()
+  add_subdirectory(omniscidb/Tests/ArrowSQLRunner)
+  add_subdirectory(omniscidb/Tests)
+endif()
+
+option(ENABLE_BENCHMARKS "Build benchmarks" ON)
+if (ENABLE_TESTS AND ENABLE_BENCHMARKS)
+  add_subdirectory(omniscidb/Benchmarks/taxi)
+endif()
+
```

**Comment by leshikus** [comment]
There was also a question from @Garra1980 why I suggest to fix HDK build related problems here. They can be fixed in HDK as well, though it would require duplicating the same code there.

**Comment by Garra1980** [comment]
> @Garra1980 asked me to add @alexbaden to the list of approvers.
> 
> Here is the patch I use to reproduce the problems in HDK:
> 
> ```
> diff --git a/CMakeLists.txt b/CMakeLists.txt
> index 0849c48..ddd5a47 100644
> --- a/CMakeLists.txt
> +++ b/CMakeLists.txt
> @@ -218,3 +218,47 @@ add_custom_target(hdk_python_clean
> 
>  add_dependencies(clean-all hdk_python_clean)
>  add_dependencies(clean-all calcite_java_clean)
> +
> +# address and thread sanitizer
> +option(ENABLE_STANDALONE_CALCITE "Require standalone Calcite server" OFF)
> +option(ENABLE_ASAN "Enable address sanitizer" OFF)
> +option(ENABLE_TSAN "Enable thread sanitizer" OFF)
> +option(ENABLE_UBSAN "Enable undefined behavior sanitizer" OFF)
> +if(ENABLE_ASAN)
> +  set(SAN_FLAGS "-fsanitize=address -O1 -fno-omit-frame-pointer")
> +  add_definitions("-DWITH_DECODERS_BOUNDS_CHECKING")
> +elseif(ENABLE_TSAN)
> +  add_definitions("-DHAVE_TSAN")
> +  # Copy the config directory to the build dir for TSAN suppressions
> +  file(COPY config DESTINATION ${CMAKE_BINARY_DIR})
> +
> +  set(SAN_FLAGS "-fsanitize=thread -fPIC -O1 -fno-omit-frame-pointer")
> +  # required for older GCC, see https://gcc.gnu.org/bugzilla/show_bug.cgi?id=64354
> +  add_definitions("-D__SANITIZE_THREAD__")
> +elseif(ENABLE_UBSAN)
> +  set(SAN_FLAGS "-fsanitize=undefined -fPIC -O1 -fno-omit-frame-pointer")
> +endif()
> +if(ENABLE_ASAN OR ENABLE_TSAN OR ENABLE_UBSAN)
> +  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${SAN_FLAGS}")
> +  set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${SAN_FLAGS}")
> +  set(ENABLE_STANDALONE_CALCITE ON)
> +endif()
> +
> +option(ENABLE_DECODERS_BOUNDS_CHECKING "Enable bounds checking for column decoding" OFF)
> +
> +if(ENABLE_STANDALONE_CALCITE)
> +  add_definitions("-DSTANDALONE_CALCITE")
> +endif()
> +
> +option(ENABLE_TESTS "Build unit tests" ON)
> +if (ENABLE_TESTS)
> +  enable_testing()
> +  add_subdirectory(omniscidb/Tests/ArrowSQLRunner)
> +  add_subdirectory(omniscidb/Tests)
> +endif()
> +
> +option(ENABLE_BENCHMARKS "Build benchmarks" ON)
> +if (ENABLE_TESTS AND ENABLE_BENCHMARKS)
> +  add_subdirectory(omniscidb/Benchmarks/taxi)
> +endif()
> +
> ```

I suppose configuration command should have some specific parameters?

**Comment by leshikus** [comment]
@Garra1980 asks if it is better adding ASAN test to omniscidb CI to instead of migrating tests to HDK

What do you think @alexbaden ?

**Review by alexbaden — APPROVED** [review]
I am fine with these changes.

As to whether we should add ASAN commands here or to HDK, it depends on what we want to test. AFAIK there is no way to run HDK without Calcite, so sanitizers won't work. If that is true, there isn't much point adding an ENABLE_ASAN option to the top-level HDK cmakelists right now.

**Pull request merged**